irohad/main/impl/storage_init.cpp

```diff
@@ -99,26 +102,29 @@ iroha::initStorage(
                 shared_model::interface::Block>>(),
             std::make_unique<shared_model::validation::ProtoBlockValidator>());
 
-    std::unique_ptr<BlockStorageFactory> temporary_block_storage_factory =
-        std::make_unique<PostgresBlockStorageFactory>(
-            pool_wrapper,
-            block_transport_factory,
-            []() { return generator::randomString(20); },
-            log_manager->getChild("TemporaryBlockStorage")->getLogger());
+    std::unique_ptr<ametsuchi::BlockStorageFactory>
+        temporary_block_storage_factory =
+            std::make_unique<ametsuchi::PostgresBlockStorageFactory>(
```

Comment:
It seems this code should not have dependency from ```postgres```. It looks better to use ```BlockStorageFactory``` interface with hidden implementation.

Reply:
This class has `const ametsuchi::PostgresOptions &postgres_options` as an argument, so I think we will need a separate class for RocksDB. How do you want to hide Postgres here exactly?